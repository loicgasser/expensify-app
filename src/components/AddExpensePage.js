import React from 'react'

const AddExpensePage = () => (
    <div>This is my expense component.</div>
)

export default AddExpensePage