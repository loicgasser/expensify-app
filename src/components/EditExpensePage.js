import React from 'react'

const EditExpensePage = (props) => (
    <div>This is my edit component. My item id is {props.id}</div>
)

export default EditExpensePage