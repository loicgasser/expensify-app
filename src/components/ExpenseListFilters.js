import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'
import { DateRangePicker } from 'react-dates'

export class ExpenseListFilters extends React.Component {
    state = {
        focusedInput: null
    }
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value)
    }
    onSortChange = (e) => {
        const value = e.target.value
        if (value === "date") {
            this.props.sortByDate()
        } else if (value === "amount") {
            this.props.sortByAmount()
        }
    }
    onDatesChange = ({ startDate, endDate }) => {
        this.props.setStartDate(startDate)
        this.props.setEndDate(endDate)
    }
    onFocusChange = (focusedInput) => {
        this.setState({ focusedInput })
    }
    render() {
        return (
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            className="text-input"
                            placeholder="Search expenses"
                            type="text"
                            value={this.props.filters.text}
                            onChange={this.onTextChange}
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            values={this.props.filters.sortBy}
                            onChange={this.onSortChange}
                        >
                            <option value="date">Date</option>
                            <option value="amount">Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate={this.props.filters.startDate}
                            startDateId="startDateId"
                            endDate={this.props.filters.endDate}
                            endDateId="endDateId"
                            onDatesChange={this.onDatesChange}
                            focusedInput={this.state.focusedInput}
                            onFocusChange={this.onFocusChange}
                            showClearDates={true}
                            numberOfMonths={1}
                            isOutsideRange={() => false}
                        />
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setTextFilter: (text) => dispatch(setTextFilter(text)),
        sortByDate: () => dispatch(sortByDate()),
        sortByAmount: () => dispatch(sortByAmount()),
        setStartDate: (startDate) => dispatch(setStartDate(startDate)),
        setEndDate: (endDate) => dispatch(setEndDate(endDate))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters)