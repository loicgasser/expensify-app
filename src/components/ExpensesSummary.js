import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import selectExpenses from '../selector/expenses';
import sumExpenses from '../selector/expenses-total'
import numeral from 'numeral'

export class ExpensesSummary extends React.Component {
    render() {
        return (
            <div className="page-header">
                <div className="content-container">
                    <h1 className="page-header__title">Viewing <span>{this.props.countExpenses}</span> expenses totalling <span>{numeral(this.props.sumExpenses / 100).format('$0,0.00')}</span></h1>
                    <div className="page-header__action">
                        <Link className="button" to="/create">Add Expense</Link>
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters)
    return {
        sumExpenses: sumExpenses(visibleExpenses),
        countExpenses: visibleExpenses.length
    }
}


export default connect(mapStateToProps)(ExpensesSummary)