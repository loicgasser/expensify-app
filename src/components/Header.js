import React from 'react'
import { connect } from 'react-redux'
import { NavLink } from 'react-router-dom'
import { startLogout } from '../actions/auth'

export const Header = (props) => (
    <header className="header">
        <div className="content-container">
            <div className="header__content">
                <NavLink className="header__title" to="/dashboard">
                    <h1>Expensify</h1>
                </NavLink>
                <button className="button button--link" onClick={props.startLogout}>Logout</button>
            </div>
        </div>
    </header>
)

const mapDispatchToProps = (dispatch) => {
    return {
        startLogout: () => dispatch(startLogout())
    }
}

export default connect(undefined, mapDispatchToProps)(Header)