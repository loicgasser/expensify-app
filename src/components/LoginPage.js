import React from 'react'
import { connect } from 'react-redux'
import { startFacebookLogin, startGoogleLogin } from '../actions/auth'

export class LoginPage extends React.Component {
    onClickFacebook = () => {
        this.props.startFacebookLogin()
    }
    onClickGoogle = () => {
        this.props.startGoogleLogin()
    }
    render() {
        return (
            <div className="box-layout">
                <div className="box-layout__box">
                    <h1 className="box-layout__title">Expensify</h1>
                    <p>It's time to get your expenses under control!</p>
                    <div className="box-layout__buttons">
                        <button className="button button--facebook" onClick={this.onClickFacebook}>Login with Facebook</button>
                        <br></br>
                        <button className="button button--google" onClick={this.onClickGoogle}>Login with Google</button>
                    </div>
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        startFacebookLogin: () => dispatch(startFacebookLogin()),
        startGoogleLogin: () => dispatch(startGoogleLogin())
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage)