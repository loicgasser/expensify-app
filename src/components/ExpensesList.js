import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import ExpenseListFilters from './ExpenseListFilters'
import getVisibleExpenses from '../selector/expenses'

export const ExpensesList = (props) => (
    <div>
        <ExpenseListFilters />
        {
            props.expenses.length === 0 ? (
                <p>No expenses</p>
            ) : (
                props.expenses.map((expense) => {
                    return <ExpenseListItem
                        key={expense.id}
                        {...expense}
                    />
                })
            )
        }
        
    </div>
)

const mapStateToProps = (state) => {
    return {
        expenses: getVisibleExpenses(state.expenses, state.filters)
    }
}

export default connect(mapStateToProps)(ExpensesList)