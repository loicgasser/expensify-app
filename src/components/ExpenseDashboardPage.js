import React from 'react'
import ExpensesList from './ExpensesList'

const ExpenseDashboardPage = () => (
    <div>
        <p>This is my dashboard component.</p>
        <ExpensesList />
    </div>
)

export default ExpenseDashboardPage