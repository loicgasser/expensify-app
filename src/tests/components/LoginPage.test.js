import React from 'react'
import { LoginPage } from '../../components/LoginPage'
import { shallow } from 'enzyme'

test('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />)
    expect(wrapper).toMatchSnapshot()
})

test('Should call startFacebookLogin on button click', () => {
    const startFacebookLogin = jest.fn()
    const wrapper = shallow(<LoginPage startFacebookLogin={startFacebookLogin}></LoginPage>)
    wrapper.find('.button--facebook').simulate('click')
    expect(startFacebookLogin).toHaveBeenCalled()
})

test('Should call startGoogleLogin on button click', () => {
    const startGoogleLogin = jest.fn()
    const wrapper = shallow(<LoginPage startGoogleLogin={startGoogleLogin}></LoginPage>)
    wrapper.find('.button--google').simulate('click')
    expect(startGoogleLogin).toHaveBeenCalled()
})