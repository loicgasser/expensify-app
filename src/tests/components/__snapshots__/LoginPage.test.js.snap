// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control!
    </p>
    <div
      className="box-layout__buttons"
    >
      <button
        className="button button--facebook"
        onClick={[Function]}
      >
        Login with Facebook
      </button>
      <br />
      <button
        className="button button--google"
        onClick={[Function]}
      >
        Login with Google
      </button>
    </div>
  </div>
</div>
`;
